cmd/ethatomicswap/main.go

```diff
@@ -0,0 +1,1364 @@
+// Copyright (c) 2018 The Decred developers and Contributors
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/bgentry/speakeasy"
+
+	ethereum "github.com/ethereum/go-ethereum"
```

Comment:
I don't like these imports, as these packages, used as libraries, are under a LGPL license.  Go will statically link against these, essentially turning the entire project GPL (it would be a GPL violation to make and distribute proprietary changes to this tool without following the terms of the GPL).

Reply:
I understand. Replacing speakeasy should be simple enough, we can even do without a library for it perhaps.

Replacing `go-ethereum` is going to be trickier, and not really something I had planned to do. Will need to to think and plan what to do about this.

Reply:
Speakeasy was never a problem, it's MIT/Apache2 licensed.  go-ethereum was the one that made me raise a few eyebrows (LGPL licensed Go packages? really?)

Reply:
This is your last remaining issue with this PR. For now I cannot however think of a solution that doesn't require me to reimplement all logic again.

Given that all sub projects are separate binaries, and given that you don't even make releases with prebuilt releases. Does it really matter that this binary becomes LGPL? Does it really affect the other binaries? You also don't embed the vendor (go-ethereum) code.

I'm not a legal person though, so perhaps it does have to impact the entire project. But as I said, there is not really a non-official go ethereum package that we could use as a replacement for the official one. I don't think anyone has ever bothered in providing that, and it seems a bit out of scope for me to start doing that.

If it really bothers you feel free to make suggestions, as I really have no clue how to make you happy with this one.

Reply:
It wouldn't affect any of the other tools, but the issue is that this repo is currently entirely liberally licensed and all imports are as well.  This is stated at the bottom of the project's readme.  If these tools are included in the repo, then it will no longer be the case where you can clone, build, and freely distribute the ethatomicswap binary without complying with the additional rules of the LGPL.

I do agree with you that rewriting libraries is out of scope for providing this tool.

Would you be ok with supporting the code in a different repository, with a description of the licensing issues associated with the tool? We can link it in this project's readme like we already do for ThreeFold.

Reply:
There was some project that was hoping to do an Apache licensed rewrite of ethereum but I think that is a long way off.

I think non-decred hosting of any tool that can't be done ISC (with preference for ISC work of course) is a great way to take care of it.

Reply:
I understand. Our `ethatomicswap` lives in <https://github.com/threefoldtech/atomicswap/tree/master/cmd/ethatomicswap> so perhaps I could close this PR, and instead make a PR to link to `ethatomicswap`. We have some documentation and documented walk-through example as well somewhere, will add it to this repo as well.

---

I hope the community will find their way to `ethatomicswap` this way though. We also already support BTC electrum wallets for atomic-swaps, planning to give such support to ethereum in the future as well.